prototype_source/gpu_direct_storage.py

```diff
@@ -0,0 +1,124 @@
+"""
+(prototype) Using GPUDirect Storage
+====================================
+
+GPUDirect Storage enabes a direct data path for direct memeory access transfers
+between GPU memory and storage, avoiding a bounce buffer through the CPU.
+
+In version ``2.7``, we introduced some prototype APIs to ``torch.cuda.gds`` that serve as thin wrappers around
+the `cuFile APIs <https://docs.nvidia.com/gpudirect-storage/api-reference-guide/index.html#cufile-io-api>`_
+that can be used with ``torch.Tensor``.
+
+In this tutorial, we will demonstrate how to use the ``torch.cuda.gds`` APIs in conjunction with
+checkpoints generated by ``torch.save`` and ``torch.load`` on local filesystem. 
+
+.. grid:: 2
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` What you will learn
+       :class-card: card-prerequisites
+
+       * Understand how to use the ``torch.cuda.gds`` APIs in conjunction with
+         checkpoints generated by ``torch.save`` and ``torch.load`` on local filesystem
+    
+    .. grid-item-card:: :octicon:`list-unordered;1em;` Prerequisites
+       :class-card: card-prerequisites
+
+       * PyTorch v.2.7.0 or later
+       * GPUDirect Storage must be installed per
+         `the documentation <https://docs.nvidia.com/gpudirect-storage/troubleshooting-guide/contents.html>`_
+       * Ensure that the filesystem that you are saving/loading to supports GPUDirect Storage.
+"""
+
+################################################################################
+# Using GPUDirect Storage with ``torch.save`` and ``torch.load``
+# =============================================================
+# GPUDirect Storage requires a storage alignment of 4KB. One can toggle this using
+# ``torch.utils.serialization.config.save.storage_alignment`` to toggle this
+
+import torch
+from torch.utils.serialization import config as serialization_config
+
+serialization_config.save.storage_alignment = 4096
+
+################################################################################
+# Given a state dictionary of tensors that are on the GPU, one can use the ``torch.serialization.skip_data`` context
```

Comment:
I think a quick overview would help here.
Using the experimental API, saving using GDS requires a few extra steps:
- Write the main checkpoint file WITHOUT any actual data. This reserves the space on disk.
- Read the offsets for each Tensor in the checkpoint using FakeTensor.
- Use GDSFile to write the appropriate data at these offsets.